### PR TITLE
refactor(query): use bigint to handle the fallback of decimal op overflow

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

use bigint to handle the fallback of decimal op overflow

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16215)

### CODE REVIEW

File: src/query/expression/src/types/decimal.rs [hunk 1]
```diff
@@ -687,19 +698,48 @@ impl Decimal for i256 {
...
+        })
     }
 
-    fn do_round_div(self, rhs: Self, mul: Self) -> Option<Self> {
-        if self.is_negative() == rhs.is_negative() {
+    fn do_round_div(self, rhs: Self, mul_scale: u32) -> Option<Self> {
+        let fallback = || {
+            let a = BigInt::from_le_bytes(&self.to_le_bytes());
+            let b = BigInt::from_le_bytes(&rhs.to_le_bytes());
+            let mul = BigInt::from(10).pow(mul_scale);
```
Comment (youngsofun) [inline src/query/expression/src/types/decimal.rs:723]: can be pre computed(maybe later)？the pow is implemented by multi mul_scale times, which is slow?

File: src/query/expression/src/types/decimal.rs [hunk 2]
```diff
@@ -725,6 +765,32 @@ impl Decimal for i256 {
...
+    fn from_bigint(value: BigInt) -> Option<Self> {
+        let mut ret: u256 = u256::ZERO;
+        let mut bits = 0;
+
+        for i in value.iter_u64_digits() {
+            if bits >= 256 {
+                return None;
+            }
+            ret |= u256::from(i) << bits;
+            bits += 64;
```
Comment (youngsofun) [inline src/query/expression/src/types/decimal.rs:777]: least significant digit first.
the last i may only used a part of 64 bits?

File: src/query/expression/src/types/decimal.rs (hunk 1 above)
Comment (sundy-li) [reply src/query/expression/src/types/decimal.rs:723]: fallback is already in a lambda function, so it's lazy computed.

File: src/query/expression/src/types/decimal.rs (hunk 2 above)
Comment (sundy-li) [reply src/query/expression/src/types/decimal.rs:777]: if `bits` is larger than 256, it's overflow, so return none.

This method is powered by https://docs.rs/num-bigint/latest/src/num_bigint/bigint/convert.rs.html#86-92

**Review by youngsofun — APPROVED** [review]

**Pull request merged**